applications/vanilla/models/class.commentmodel.php

```diff
@@ -1129,8 +1129,11 @@ public function save($formPostValues, $settings = false) {
             if (!$insert || !$this->checkUserSpamming(Gdn::session()->UserID, $this->floodGate)) {
                 $fields = $this->Validation->schemaValidationFields();
                 unset($fields[$this->PrimaryKey]);
+                $comment = $this->getID($commentID, DATASET_TYPE_ARRAY);
+                $insertUserID = $comment['InsertUserID'] ?? null;
```

Comment:
Can't you just unset InsertUserID from `$fields` and not fetch the record?

Reply:
InsertUserID is not set on the $fields array we need to add it before firing AfterValidateComment.